client/gtk2/ibusimcontext.c

```diff
@@ -512,43 +486,39 @@ _process_key_event (IBusInputContext *context,
         break;
     }
     case 2: {
-        GSource *source = g_timeout_source_new (1);
-        ProcessKeyEventReplyData *data = NULL;
-
-        if (source)
-            data = g_slice_new0 (ProcessKeyEventReplyData);
-        if (!data) {
-            g_warning ("Cannot wait for the reply of the process key event.");
-            retval = ibus_input_context_process_key_event (context,
-                                                           keyval,
-                                                           keycode - 8,
-                                                           state);
-            if (source)
-                g_source_destroy (source);
-            break;
-        }
-        data->count = 1;
-        g_source_attach (source, NULL);
-        g_source_unref (source);
-        data->count_cb_id = g_source_get_id (source);
+        ProcessKeyEventSyncData data;
+
+        data.res = NULL;
+        data.context = g_main_context_new ();
+        data.loop = g_main_loop_new (data.context, FALSE);
```

Comment:
Probably I think if data.context or data.loop is null, _process_key_event_sync() could be called?

Reply:
Okay, I will update the code.